models/DataObject/Classificationstore.php

```diff
@@ -198,14 +198,18 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
 
         if (!$this->isFieldDirty('_self')) {
             if ($this->object) {
-                $oldData = $this->items[$groupId][$keyId][$language];
-                $oldData = $dataDefinition->getDataForResource($oldData, $this->object);
-                $oldData = serialize($oldData);
+                $oldData = null;
+                
+                if (!isset($this->items[$groupId][$keyId][$language])) {
```

Comment:
remove the if

now we compare data which is transformed by getDataForResource with data that isn't